classes/ShopMaintenance.php

```diff
@@ -118,4 +125,125 @@ public static function cleanAdminControllerMessages()
             }
         }
     }
+    
+     /**
+     * Clean image products directories 'img/p'.
+     *
+     * @since 1.0.9
+     */
+    public static function CheckAndFixImages($imagesNow,$imagesMax)
+    {
+		$starttime = time(); 
+		$path = _PS_ROOT_DIR_.'/img/p/';
+		$directory = new \RecursiveDirectoryIterator($path);
+		$iterator = new \RecursiveIteratorIterator($directory);
+		$count_ko = 0 ;	
+		$count_noexist = 0;
+					
+		if ($imgtypes = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('SELECT name FROM '._DB_PREFIX_.'image_type WHERE products= 1')) {
+			foreach ($imgtypes as $type)
+			$Image_types_product[]=$type['name'];
+		}
+
+		if ($images_list = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('SELECT id_image FROM '._DB_PREFIX_.'image_shop WHERE 1')) {
+			foreach ($images_list as $img)
+			$Image_listing[]=$img['id_image'];
+		}
+
+		foreach ($iterator as $info) {
+			$filename = $info->getFilename();
+			$pathname = $info->getPathname();
+			   
+			if ($info->isDir() || $filename=='index.php'){ 
+				 continue;
+			}
+			$img_nbr = explode("-", $filename, 2);
+			$img_nbr[0] = str_replace(".jpg", "", $img_nbr[0]);
+
+			if (isset($img_nbr[1])) {
+				$img_type_name = str_replace(".jpg", "", $img_nbr[1]);
+				$images[] = array('id' => $img_nbr[0], 'path' => $pathname, 'type' => $img_type_name);
+			}
+			else {
+				$images[] = array('id' => $img_nbr[0], 'path' => $pathname, 'type' => 'default');
+			}
+		}
+			
+		$imagesMax = count($images)-1;
+					
+		foreach($images as $key => $value) {
+			if ($key < $imagesNow) continue;
+			$image_id = intval($value['id']);
+			if (!in_array($image_id, $Image_listing, true)) {
+				$count_ko++;
+				unlink($value['path']);
+			}
+			elseif (!in_array($value['type'], $Image_types_product, true) && $value['type']!='default') {
+				$count_ko++;
+				unlink($value['path']);
+			}
```

Comment:
I miss deleting the entire image directory here for images which no longer exist. For deleting recursively, there's `Tools::deleteDirectory()`.

Reply:
I'll implemant this corectly then, nice move !